### PR TITLE
Add Tags to Writing Settings in SiteSettingsViewController

### PR DESCRIPTION
**Pull request by ctarda**
Another step in the implementation of #8232 

In this PR:
- Add a row to the Writing Section of SiteSettingsViewController.
- This row would be "editable" only for site admins
- Update the signature of `syncTagsForBlog` in `PostTagService` in order to return the list of Tags. It kind of makes sense to me, in order to make it symmetric with the rest of the API and it does not seem to break anything.
- Get the number of tags from the service, attacking `syncTagsForBlog`. In a first draft, I was using `getTopTagsForBlog`, but the name seems to imply that I might not get all the tags, and the implementation is paginated, so I decided to go with `syncTagsForBlog`. I am not sure if that is the right way to go.

Things missing:
- Nothing happens if you tap the new cell. One thing at a time... ;) Trying to keep PRs short 😛 

Pinging @aerych

### CODE REVIEW

**Comment by ctarda** [comment]
Ping @aerych 

Please check the last two commits. At first, I was loading the tags from the PostTagService, but I've noticed the Blog object has a tags property. So in the very last commit I've switched to use that relationship to get the number of tags. I am not sure what's the right way to do it though. 🤷‍♂️

File: WordPress/Classes/Services/PostTagService.m [hunk 1]
```diff
@@ -13,8 +13,8 @@ @interface PostTagService ()
 @implementation PostTagService
 
 - (void)syncTagsForBlog:(Blog *)blog
-                success:(nullable void (^)(void))success
-                failure:(nullable void (^)(NSError *error))failure
+                success:(nullable void (^)(NSArray <PostTag *> *tags))success
+                failure:(nullable void (^)(NSError *error))failure;
```
Comment (aerych) [inline WordPress/Classes/Services/PostTagService.m]: Looks like a stray semicolon might have sneaked in here.

File: WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m [hunk 2]
```diff
@@ -974,6 +997,14 @@ - (void)loginValidationFailedWithError:(NSError *)error
     }
 }
 
+- (NSString *)getTagsCountPresentableString:(NSInteger)tagCount
+{
+    NSString *format = tagCount == 1 ? NSLocalizedString(@"%@ Tag", @"The number of tags in the writting settings. Singular.") : NSLocalizedString(@"%@ Tags", @"The number of tags in the writting settings. Plural.");
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m]: Our translators might not understand the significance of the `%@`.  Maybe mention its a placeholder in the description.

I dig the user of the trinary operator, but the line is getting pretty long, so its not too readable at a glance.  Could we make a change to improve readability?

File: WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m [hunk 3]
```diff
@@ -347,6 +360,12 @@ - (void)configureDefaultCategoryCell
     [self.defaultCategoryCell setTextValue:[postCategory categoryName]];
 }
 
+- (void)configureTagsCell
+{
+    NSInteger tagCount = self.blog.tags.allObjects.count;
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m]: 👍

File: WordPress/Classes/Services/PostTagService.m (hunk 1 above)
Comment (ctarda) [reply WordPress/Classes/Services/PostTagService.m]: Ouch. Old habits die hard... It's a little thing I starting doing years ago (if you copy the method declaration from the .h file, you ca just paste it i the .m file as it is). Sorry, I will remove it right away!

File: WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m (hunk 2 above)
Comment (ctarda) [reply WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m]: I tried to reformat the ternary operator, but I was not convinced, so I did a traditional if/else. You can check the last two commits, see which version looks more readable to you.

**Comment by ctarda** [comment]
Updated! Ping @aerych

File: WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m [hunk 4]
```diff
@@ -974,6 +997,19 @@ - (void)loginValidationFailedWithError:(NSError *)error
...
 }
 
+- (NSString *)getTagsCountPresentableString:(NSInteger)tagCount
+{
+    NSString *format;    
+    if (tagCount == 1) {
+        format = NSLocalizedString(@"%@ Tag", @"The number of tags in the writting settings. Singular. %@ is a placeholder for the number");
+    } else {
+        format = NSLocalizedString(@"%@ Tags", @"The number of tags in the writting settings. Plural. %@ is a placeholder for the number");
+    }
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m]: I know I'm being nitpicky, but we could avoid the else clause if one of the strings was used to initialize `format`

Comment (ctarda) [reply WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m]: Done! :)

Comment (aerych) [reply WordPress/Classes/ViewRelated/Blog/SiteSettingsViewController.m]: 💥 !

**Comment by aerych** [comment]
Awesome @ctarda :)  :shipit:

**Comment by ctarda** [comment]
\o/

**Pull request merged**